x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/Expressions.java

```diff
@@ -82,7 +82,13 @@ public static AttributeSet references(List<? extends Expression> exps) {
     }
 
     public static String name(Expression e) {
-        return e instanceof NamedExpression ? ((NamedExpression) e).name() : e.nodeName();
+        if (e instanceof NamedExpression) {
```

Comment:
I feel like we'd be better off being more OO here and adding a `colunmName` function to `Expression` or something like that. The fix is fine with me as it stands but I think something like that would make a good follow up change.

Reply:
This is a good example of code exposing conceptual leaks. You are right that this needs OO-ing. There's already a class for that `NamedExpression` is just that `Literal`s are not yet one. Which is the culprit - I'll raise an issue to chase that down and make sure it sits well in the overall structure.

Reply:
Raised https://github.com/elastic/elasticsearch/issues/33523